tests/conftest.py

```diff
@@ -1650,27 +1652,93 @@ def collect_db_dump(request, duthosts):
+            duts_data[duthost]["config"] = {"failed": False, "config_differences":None}
+
+            if "20191130" in duthost.os_version:
+                pre_existing_core_dumps = duthost.shell('ls /var/core/ | grep -v python')['stdout'].split()
+            else:
+                pre_existing_core_dumps = duthost.shell('ls /var/core/')['stdout'].split()
+            duts_data[duthost]["pre_core_dumps"] = pre_existing_core_dumps
 
-    yield
-    if "20191130" in duthost.os_version:
-        cur_cores = duthost.shell('ls /var/core/ | grep -v python')['stdout'].split()
-    else:
-        cur_cores = duthost.shell('ls /var/core/')['stdout'].split()
+            # get running config before running
+            duts_data[duthost]["pre_running_config"] = json.loads(duthost.shell("sonic-cfggen -d --print-data", verbose=False)['stdout'])
 
-    new_core_dumps = set(cur_cores) - set(pre_existing_cores)
-    # convert to list so print msg will not contain "set()"
-    new_core_dumps = list(new_core_dumps)
+    yield
 
-    if new_core_dumps:
-        pytest.fail("Core dumps found. Expected: %s Found: %s. Test failed. New core dumps: %s" % (len(pre_existing_cores),\
-            len(cur_cores), new_core_dumps))
+    if check_flag:
+        for duthost in duts_data:
+            if "20191130" in duthost.os_version:
+                cur_cores = duthost.shell('ls /var/core/ | grep -v python')['stdout'].split()
+            else:
+                cur_cores = duthost.shell('ls /var/core/')['stdout'].split()
+            duts_data[duthost]["cur_core_dumps"] = cur_cores
+
+            new_core_dumps = set(duts_data[duthost]["cur_core_dumps"]) - set(duts_data[duthost]["pre_core_dumps"])
+            # convert to list so print msg will not contain "set()"
+            new_core_dumps = list(new_core_dumps)
+
+            if new_core_dumps:
+                duts_data[duthost]["core_dump"]["failed"] = True
+                duts_data[duthost]["core_dump"]["core_dump_files"] = new_core_dumps
+                logger.info("Core dumps found. Expected: %s Found: %s. New core dumps: %s" % (len(pre_existing_core_dumps), len(cur_cores), new_core_dumps))
+
+            # get running config before running
+            duts_data[duthost]["cur_running_config"] = json.loads(duthost.shell("sonic-cfggen -d --print-data", verbose=False)['stdout'])
+
+            different_config_pre = {}
+            different_config_cur = {}
+
+            # Check if the running config is modified after the test case running
+            for key, value in duts_data[duthost]["cur_running_config"].items():
+                # skip check "FLEX_COUNTER_TABLE" due to pr https://github.com/Azure/sonic-buildimage/pull/10413
+                if key in ["FLEX_COUNTER_TABLE", "VLAN_INTERFACE", "PORT", "AAA", "TACPLUS", "RADIUS", "TACPLUS_SERVER", "FEATURE"]:
+                    continue
+                if duts_data[duthost]["pre_running_config"][key] != value:
```

Comment:
check if key is present in dict before accessing the value?
This is causing an error in multi-asic testbed.
>                   if duts_data[duthost]["pre_running_config"][key] != value:
E                   KeyError: u'KDUMP'

Reply:
> 

Hi, sucarna, this is a test case issue, during the test case running, it adds the key 'KDUMP' in config, and doesn't delete it when finish running, I will fix it later.

Reply:
Is that deletion fixed in this PR or will there be a follow up PR? Why is it that it was failing only on multi-asic testbed?